src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,129 @@
+        User user = new User("user_with_min_pass_length", "123456", MIN_AGE);
+        User actualUser = registrationService.register(user);
+        assertEquals(actualUser, user);
+    }
+
+    @Test
+    void register_spaceLogin_notOk() {
+        User user = new User(" ", PASSWORD, MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Invalid data"));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User user = new User("", PASSWORD, MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Invalid data"));
+    }
+
+    @Test
+    void register_passwordWithSpace_notOk() {
+        User user = new User(LOGIN, "     d", MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains(PASSWORD));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User user = new User(LOGIN, PASSWORD, MIN_AGE);
+        User correctUser = registrationService.register(user);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(correctUser),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains(correctUser.getLogin()));
+    }
+
+    @Test
+    void register_passwordLessThanSix_notOk() {
+        User user = new User(LOGIN, "12345", MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains(PASSWORD));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User(LOGIN, null, MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains(PASSWORD));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User(null, PASSWORD, MIN_AGE);
+        RuntimeException thrown = assertThrows(RuntimeException.class,
+                () -> registrationService.register(user),
+                "Expected register() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Invalid data"));
+    }
+
+    @Test
+    void register_ageLessThanEighteen_notOk() {
```

Comment:
```suggestion
    void register_ageLessThanRequiredMin_notOk() {
```
don't put specific values int method signature, required age can be changed to another number

Reply:
fixed